### PR TITLE
Simplify case report loading commands and add RNA multiqc report loading

### LINKED ISSUE #3647: RNA multiQC report
As discussed during a meeting on 22-10-28 with @dnil and @northwestwitch, it would be good for the customers who has RNA data associated with a DNA case to also have access to the MultiQC report for the RNA sequences.

Adding it as a report, perhaps just differentiating it from the DNA multiQC with some added tag would be enough for now.

### PR DESCRIPTION
**Pull request by northwestwitch**
- Load an RNA multiqc for a case manually with command `scout load report -t multiqc_rna` (fix #3647 and fix partially #3648) 
- Load RNA multiqc report automatically for a case if its config file contains the `multiqc_rna` key/value
- Simplifies loading of all reports by refactoring the associated code. Specifically:
   - Deprecating CLI commands `scout load <delivery_report, gene_fusion_report, coverage_qc_report, cnv_report>`
   - Replacing them with command: `scout load report -t <report type>`

BONUS feature:
- Update multiqc file for a case using the command `scout load report -t multiqc`

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the paxa software installed on `hasta`:
 - Log in into hasta: `ssh <USER.NAME>@hasta.scilifelab.se`
 - Activate the staging environment with the command `us` (Use Stage)
 - Run the command `paxa` and follow the instructions. Make sure to specify scout-stage as the resource you request and the server cg-vm1 as server.
1. Log out from the hasta server.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, log out from `cg-vm1` and log in again in the `hasta` server, repeat the `hasta` and `paxa` procedure, which will release the allocated resource (scout-stage) to be used for testing by other users.
</details>

**How to test -> locally**:
1. Try to **load/update the 5 existing types of reports with the usual commands `case load`**:
   - case load cnv-report
   - case load coverage-qc-report
   - case load delivery-report
   - case load gene-fusion-report (--research) # it's 2 distinct reports for clinical and research
  
All these command should continue to work but should raise a deprecation message

2. **Load/update the same reports as before using the new command: scout load report <report-type>**

All 5 reports should be loaded and visible on case page

3. **Try loading the RNA multiqc report (any HTML file will do) with the same command as in 2.**

This report should be also visible on case page sidebar

4. **Testing of bonus feature: try to load/update a multiqc report** (any HTML file will do) with the command: `scout load report -t multiqc`

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [ ] code approved by
- [x] tests executed by

### CODE REVIEW

File: scout/load/report.py [hunk 1]
```diff
@@ -1,159 +1,26 @@
...
 
-from scout.adapter import MongoAdapter
+from scout.constants import CUSTOM_CASE_REPORTS
 from scout.exceptions import DataNotFoundError, IntegrityError
+from scout.server.extensions import store
 
 LOG = logging.getLogger(__name__)
 
 
-def load_delivery_report(
```
Comment (northwestwitch) [inline scout/load/report.py:10]: These methods are a lot repetitive and can be replaced by a few lines of code instead, see at the bottom of this file

File: scout/server/blueprints/cases/views.py [hunk 2]
```diff
@@ -699,7 +699,7 @@ def gene_fusion_report(institute_id, case_name, report_type):
     """Download gene fusion report"""
     _, case_obj = institute_and_case(store, institute_id, case_name)
     report_path = case_obj.get(report_type)
-    if report_path is None or report_type not in CUSTOM_CASE_REPORTS:
+    if report_path is None or report_type not in CUSTOM_CASE_REPORTS.values():
```
Comment (dnil) [inline scout/server/blueprints/cases/views.py:702]: 👍

File: scout/constants/case_tags.py [hunk 3]
```diff
@@ -1,12 +1,14 @@
...
-    "coverage_qc_report",
-    "gene_fusion_report",
-    "gene_fusion_report_research",
-]
+CUSTOM_CASE_REPORTS = {
+    "delivery": "delivery_report",
+    "cnv": "cnv_report",
+    "cov_qc": "coverage_qc_report",
+    "multiqc": "multiqc",
+    "multiqc_rna": "multiqc_rna",
```
Comment (northwestwitch) [inline scout/constants/case_tags.py:8]: This is the new one!

File: tests/load/test_load_report.py [hunk 4]
```diff
@@ -1,114 +0,0 @@
...
-    ## WHEN trying to load a report for a case_id that does not exist in the data base
-    case_id = "id_of_non_existing_case"
-    report_path = "a_fakey_path"
-
-    ## THEN an exception should be raised
-    with pytest.raises(DataNotFoundError):
-        load_delivery_report(adapter=adapter, case_id=case_id, report_path=report_path)
-
-
-def test_load_delivery_report_using_case_id_without_update_fail(adapter, case_obj):
```
Comment (dnil) [inline tests/load/test_load_report.py:21]: Couldn't we refactor at least one test with an expected fail?

File: scout/commands/load/report.py [hunk 5]
```diff
@@ -4,87 +4,61 @@
...
-from scout.server.extensions import store
+from scout.constants import CUSTOM_CASE_REPORTS
+from scout.load.report import update_case_report
 
 LOG = logging.getLogger(__name__)
 
 
-@click.command("delivery-report")
+####### The following commands are deprecated and will be removed in Scout release >= 5.0 #######
+@click.command("delivery-report", deprecated=True)
```
Comment (dnil) [inline scout/commands/load/report.py:14]: 🥇

**Comment by northwestwitch** [comment]
> Generally speaking it is also a little premature to remove the tests for the deprecated functions, doesn't it?

I didn't, I've just removed tests for the scout.load functions (which were already redundant). The tests for the commands.load functions are still  there:
- [test load cnv report](https://github.com/Clinical-Genomics/scout/blob/main/tests/commands/load/test_load_cnv_report_cmd.py)
- [test load coverage qc report](https://github.com/Clinical-Genomics/scout/blob/main/tests/commands/load/test_load_coverage_qc_report_cmd.py)
- [test load delivery report](https://github.com/Clinical-Genomics/scout/blob/main/tests/commands/load/test_load_delivery_report_cmd.py)
- [test load fusion report and fusion report research](https://github.com/Clinical-Genomics/scout/blob/main/tests/commands/load/test_load_report_cmd.py)

My idea was to keep these tests to make sure that the usage of the old commands still work. And when we remove the deprecated commands we can rewrite these tests to use the new commands, no?

File: scout/constants/case_tags.py (hunk 3 above)
Comment (northwestwitch) [reply scout/constants/case_tags.py:8]: Told you that it's confusing!

**Comment by dnil** [comment]
> > Generally speaking it is also a little premature to remove the tests for the deprecated functions, doesn't it?
> 
> I didn't, I've just removed tests for the scout.load functions (which were already redundant). The tests for the commands.load functions are still there:
> 
> * [test load cnv report](https://github.com/Clinical-Genomics/scout/blob/main/tests/commands/load/test_load_cnv_report_cmd.py)
> * [test load coverage qc report](https://github.com/Clinical-Genomics/scout/blob/main/tests/commands/load/test_load_coverage_qc_report_cmd.py)
> * [test load delivery report](https://github.com/Clinical-Genomics/scout/blob/main/tests/commands/load/test_load_delivery_report_cmd.py)
> * [test load fusion report and fusion report research](https://github.com/Clinical-Genomics/scout/blob/main/tests/commands/load/test_load_report_cmd.py)
> 
> My idea was to keep these tests to make sure that the usage of the old commands still work. And when we remove the deprecated commands we can rewrite these tests to use the new commands, no?

Ah, right! That works for me!

File: tests/load/test_load_report.py (hunk 4 above)
Comment (northwestwitch) [reply tests/load/test_load_report.py:21]: I've removed the --update option so the command scout load report <whichever-report> will always update case and never fail. I was reasoning that if you use the command you probably want to update the report anyway. And if you update it wrongly you can always re-update it later

File: tests/load/test_load_report.py [hunk 6]
```diff
@@ -1,114 +0,0 @@
-import pytest
-
-from scout.exceptions import DataNotFoundError, IntegrityError
-from scout.load.report import load_cnv_report, load_coverage_qc_report, load_delivery_report
-
-
-def test_load_delivery_report_bad_case_id(adapter):
```
[the remark this answers is not in the document]
Comment (northwestwitch) [reply tests/load/test_load_report.py:7]: I've modified the test for the new command to use the @pytest.mark.parametrize("report_types"), so it is testing the loading of all report types now:

image

**Comment by northwestwitch** [comment]
Added the refactoring of the scout load report cli to the to-do list of #2357, so we remember to fix this when we release a major. Merging now!

**Pull request merged**